### PR TITLE
Regression test for 91 (global root mismatch for block 47047)

### PR DESCRIPTION
**Pull request by CHr15F0x**
With the current shape of `update()` I can assert for the result of the update, which previously would yield global root of zero (ie, empty tree).

### CODE REVIEW

File: crates/pathfinder/src/state.rs [hunk 1]
```diff
@@ -824,18 +828,137 @@ mod tests {
...
+        // We need to set the magic bytes for zstd compression to simulate a compressed
+        // contract definition, as this is asserted for internally
+        let zstd_magic = vec![0x28, 0xb5, 0x2f, 0xfd];
+
+        let contract_fetched_compressed = FetchedCompressedContract {
+            deploy_info: DeployedContract {
+                address: contract_addr,
+                call_data: vec![],
+                hash: contract_hash,
+            },
```
Comment (koivunej) [inline crates/pathfinder/src/state.rs:918]:
```suggestion
            deploy_info: contract_deploy,
```

File: crates/pathfinder/src/state.rs [hunk 2]
```diff
@@ -620,6 +620,8 @@ fn update(
         .apply()
         .context("Applying global state tree updates")?;
 
+    eprintln!("{:X}", new_global_root.0);
```
Comment (CHr15F0x) [inline crates/pathfinder/src/state.rs:623]: Whoopsie, a debug to be removed.

**Review by koivunej — APPROVED** [review]
I think the one deployedcontract should be deduplicated because it should be the same, but other than that good to go.

**Comment by Mirko-von-Leipzig** [comment]
Closes #91

**Pull request merged**